packages/deployer/index.js

```diff
@@ -18,9 +18,11 @@ class Deployer extends Deployment {
     this.provider = options.provider;
     this.known_contracts = {};
     if (options.ens && options.ens.enabled) {
-      options.ens.registryAddress = this.networks[this.network].registry
-        ? this.networks[this.network].registry.address
-        : null;
+      options.ens.registryAddress =
+        this.networks[this.network].registryAddress ??
+        this.networks[this.network].registry?.address ??
+        options.ens.registryAddress ??
+        options.ens.registry?.address;
```

Comment:
As you mentioned, this is not ideal. I think it would be better to have Config resolver fn `ensRegistryAddress(networkName)`

Reply:
A config resolver function?  What does that mean?  (Just to be clear what I was talking about above having just a getter that returns the *current* registry address, it wouldn't take any arguments.)

Reply:
ah shoot, **_CURRENT!_** I forgot it gets set, and the getter won't need a network name.